### PR TITLE
Unify back::msl push constants map and sizes buffer

### PR DESCRIPTION
**Pull request by lachlansneff**
This allows the sizes buffer slots to be tightly packed.

### CODE REVIEW

File: src/back/msl/mod.rs [hunk 1]
```diff
@@ -71,13 +71,27 @@ pub type BindingMap = std::collections::BTreeMap<BindSource, BindTarget>;
 #[derive(Clone, Debug, Default, Hash, Eq, Ord, PartialEq, PartialOrd)]
 #[cfg_attr(feature = "serialize", derive(serde::Serialize))]
 #[cfg_attr(feature = "deserialize", derive(serde::Deserialize))]
-pub struct PushConstantsMap {
+pub struct PerStageResources {
```
Comment (kvark) [inline src/back/msl/mod.rs:74]: excellent!

File: src/back/msl/mod.rs [hunk 2]
```diff
@@ -282,6 +293,32 @@ impl Options {
             None => Err(EntryPointError::MissingPushConstants(stage)),
         }
     }
+
+    fn resolve_sizes_buffer(
+        &self,
+        stage: crate::ShaderStage,
+    ) -> Result<ResolvedBinding, EntryPointError> {
+        let slot = match stage {
+            crate::ShaderStage::Vertex => self.per_stage_map.vs.sizes_buffer,
```
Comment (kvark) [inline src/back/msl/mod.rs:302]: if you feel fancy, you could implement `Index<crate::ShaderStage> for PerStageMap`

**Review by kvark — APPROVED** [review]
Wonderful!

File: src/back/msl/mod.rs (hunk 2 above)
Comment (lachlansneff) [reply src/back/msl/mod.rs:302]: That's a good idea, I'll do that.

**Pull request merged**